src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeStartUpTests.cs

```diff
@@ -0,0 +1,215 @@
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.EslintBridgeClient
+{
+    [TestClass]
+    public class EslintBridgeStartUpTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<EslintBridgeStartUp, IEslintBridgeStartUp>(null, new[]
+            {
+                MefTestHelpers.CreateExport<string>("some path", EslintBridgeStartUp.EslintBridgeDirectoryMefContractName),
+                MefTestHelpers.CreateExport<INodeLocator>(Mock.Of<INodeLocator>()),
+                MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>())
+            });
+        }
+
+        [TestMethod]
+        public void Start_FailsToFindNode_FileNotFoundException()
+        {
+            var nodeLocator = SetupNodeLocator(null);
+
+            var testSubject = CreateTestSubject(nodeLocator: nodeLocator.Object);
+            Func<Task> act = async () => await testSubject.Start();
+
+            act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain("node.exe");
+        }
+
+        [TestMethod]
+        public void Start_FailsToFindStartupScript_FileNotFoundException()
+        {
+            const string filePath = "somefile.txt";
+
+            var fileSystem = SetupStartupScriptFile(filePath, false);
+            var testSubject = CreateTestSubject(filePath, fileSystem: fileSystem.Object);
+            Func<Task> act = async () => await testSubject.Start();
+
+            act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain(filePath);
+        }
+
+        [TestMethod]
+        public async Task Start_ServerStarts_ReturnsPortNumber()
+        {
+            var scriptPath = CreateScriptThatPrintsPortNumber(123);
+            var startupScriptPath = " "; // pass whitespace so it would not fail NullOrEmpty validation but also do nothing
+
+            var nodeLocator = SetupNodeLocator(scriptPath);
+            var testSubject = CreateTestSubject(startupScriptPath, nodeLocator: nodeLocator.Object);
+
+            var port = await testSubject.Start();
+            port.Should().Be(123);
+        }
+
+        [TestMethod]
+        public async Task Start_TaskSucceeds_NextCallDoesNotStartAnotherServer()
+        {
+            var scriptPath = CreateScriptThatPrintsPortNumber(123);
+            var startupScriptPath = " "; // pass whitespace so it would not fail NullOrEmpty validation but also do nothing
+
+            var nodeLocator = SetupNodeLocator(scriptPath);
+            var testSubject = CreateTestSubject(startupScriptPath, nodeLocator: nodeLocator.Object);
+
+            var port = await testSubject.Start();
+            port.Should().Be(123);
+
+            port = await testSubject.Start();
+            port.Should().Be(123);
+
+            nodeLocator.Verify(x => x.Locate(), Times.Once);
```

Comment:
Not the most reliable way to test this but I don't see another option besides wrapping Process.Start :/